src/promptflow/promptflow/_core/flow_execution_context.py

```diff
@@ -141,6 +142,14 @@ async def invoke_tool_async(self, node: Node, f: Callable, kwargs):
             self._run_tracker.end_run(node_run_id, result=result, traces=traces)
             flow_logger.info(f"Node {node.name} completes.")
             return result
+        # User tool should reraise the CancelledError after its own handling logic,
+        # so that the error can propagate to the scheduler for handling.
+        # Otherwise, the node would end with Completed status.
+        except asyncio.CancelledError as e:
```

Comment:
So, if user code reraise the `CancelledError`, it will not be wrapped as `ToolExecutionError`?
![image](https://github.com/microsoft/promptflow/assets/17527303/7ddce7ee-a056-46b1-ad5e-91d4c7827503)

Reply:
Good question.
It will not be wrapped by ToolExecutionError, and CanceledError will be specially handled in error response creation, so that the error response will be null.